### ISSUE TITLE
Adding rows to Table not working when sheet contains multiple tables of different widths

### ISSUE THREAD

**Issue by craigbrown** [EPPlusSoftware/EPPlus#423]
I have found a couple of bugs that occur when a sheet contains multiple tables, one below the other, and those tables take up different numbers of columns. I don't know if these are separate bugs but hopefully it's okay to group them under one issue since they are related.

### 1. Can't add row to table if it is below a wider table

I have two tables on a worksheet. The top table covers more columns than the bottom table.

<img width="624" alt="Screenshot 2021-06-29 at 14 47 26" src="https://user-images.githubusercontent.com/8934235/123810400-44042000-d8ea-11eb-9f39-0079e964ae81.png">

If I try to add 10 rows to the top table, this works fine – the top table has 10 extra rows, and the bottom table is moved down by 10 rows. But if I try to add 10 rows to the bottom table, I get an `InvalidOperationException` with the message `Can't insert into the range. Cells collide with table Table1`. I don't see why adding rows to the end of the bottom table would affect the top table at all.

**Unit Test**
```csharp
[TestMethod]
public void AddRowsToTablesOfDifferentWidths_TopWider()
{
    using (var pck = OpenTemplatePackage("TestTableAddRows.xlsx"))
    {
        // Get sheet 1 from the workbook, and get the tables we are going to test
        var ws = TryGetWorksheet(pck, "Sheet1");
        var table1 = ws.Tables["Table1"];
        var table2 = ws.Tables["Table2"];
        // Make sure the tables are where we expect them to be
        if (table1.Address.ToString() != "B2:E3") Assert.Inconclusive();
        if (table2.Address.ToString() != "B6:C7") Assert.Inconclusive();

        // Add 10 rows to Table1
        table1.AddRow(10);
        // Make sure Table1's address has been correctly updated
        Assert.AreEqual("B2:E13", table1.Address.ToString());
        // Make sure Table2 below has been correctly moved
        Assert.AreEqual("B16:C17", table2.Address.ToString());

        // Add 10 rows to Table2
        table2.AddRow(10);
        // Make sure Table2 has been correctly updated
        Assert.AreEqual("B16:C27", table2.Address.ToString());
        // Make sure Table1 hasn't moved
        Assert.AreEqual("B2:E13", table1.Address.ToString());
    }
}
```

### 2. Adding row to table doesn't move wider table below it

<img width="627" alt="Screenshot 2021-06-29 at 15 08 14" src="https://user-images.githubusercontent.com/8934235/123815153-2638ba00-d8ee-11eb-9d72-4628a178d06b.png">

Now if I have the tables the other way around, I have a different problem.

If I try to add 10 rows to the top table, its range is correctly extended. But the bottom table is not moved down at all. So the address of the bottom table is now in the middle of the top table. 

This doesn't happen when both tables are the same width, or when the bottom table is narrower.

**Unit Test**
```csharp
[TestMethod]
public void AddRowsToTablesOfDifferentWidths_BottomWider()
{
    using (var pck = OpenTemplatePackage("TestTableAddRows.xlsx"))
    {
        // Get sheet 2 from the workbook, and get the tables we are going to test
        var ws = TryGetWorksheet(pck, "Sheet2");
        var table3 = ws.Tables["Table3"];
        var table4 = ws.Tables["Table4"];
        // Make sure the tables are where we expect them to be
        if (table3.Address.ToString() != "B2:C3") Assert.Inconclusive();
        if (table4.Address.ToString() != "B6:E7") Assert.Inconclusive();

        // Add 10 rows to Table3
        table3.AddRow(10);
        // Make sure Table3's address has been correctly updated
        Assert.AreEqual("B2:C13", table3.Address.ToString());
        // Make sure Table4 below has been correctly moved
        Assert.AreEqual("B16:E17", table4.Address.ToString());

        // Add 10 rows to Table4
        table4.AddRow(10);
        // Make sure Table4 has been correctly updated
        Assert.AreEqual("B16:E27", table4.Address.ToString());
        // Make sure Table3 hasn't moved
        Assert.AreEqual("B2:C13", table3.Address.ToString());
    }
}
```
---
[TestTableAddRows.xlsx](https://github.com/EPPlusSoftware/EPPlus/files/6734303/TestTableAddRows.xlsx)

**Comment by JanKallman** [comment]
Thanks, I'll have a look

**Comment by JanKallman** [comment]
I have added a fix for the first test. The second test will throw an exception as the top table is smaller than the bottom one and will shift a part of the columns in the second table.

**Comment by craigbrown** [comment]
Ah okay I see, I thought `AddRow` was adding an entire row to the worksheet. Thanks for fixing the first one though.

In the second example, is there any way to extend that table with EPPlus? I was hoping I could do something like
```csharp
ws.InsertRow(3, 10);
table3.Resize(10);
```
but there's no method I can see that lets me manually change the size of a table.
Is there a way of doing this or is that something you could consider adding?

**Comment by JanKallman** [comment]
table.AddRow only add rows to the table.
`ws.InsertRow(3, 10);` should insert 10 rows to the table and shift the other table down.  Or user Range.Insert

**Comment by craigbrown** [comment]
Ah yes of course! Many thanks.

**Comment by JanKallman** [comment]
Fixed in version 5.7.1
